ts/client/src/accounts/mangoAccount.ts

```diff
@@ -1342,15 +1346,40 @@ export class PerpPosition {
     );
   }
 
-  public getPnl(perpMarket: PerpMarket): I80F48 {
+  public cumulativePnlOverPositionLifetimeUi(
+    group: Group,
+    perpMarket: PerpMarket,
+  ): number {
     if (perpMarket.perpMarketIndex !== this.marketIndex) {
       throw new Error("PerpPosition doesn't belong to the given market!");
     }
 
-    return this.quotePositionNative.add(
-      I80F48.fromI64(this.basePositionLots.mul(perpMarket.baseLotSize)).mul(
-        perpMarket.price,
+    const priceChange = perpMarket.price.sub(
+      this.getAverageEntryPrice(perpMarket),
+    );
+
+    return toUiDecimals(
```

Comment:
could be worth to add a `perpMarket.settleTokenNativeToUi()`

Reply:
I will do this later, its a lightly larger change I realise now